src/lib.rs

```diff
@@ -395,66 +193,34 @@ enum Case {
 ///
 /// This method is intended for implementing traits from [`std::fmt`].
 ///
-/// # Errors
+/// # Deviations from standard.
 ///
-/// * Deviations from standard.
 /// * No length limits are enforced for the data part.
 pub fn encode_to_fmt<T: AsRef<[u5]>>(
     fmt: &mut dyn fmt::Write,
-    hrp: Hrp,
-    data: T,
-    variant: Variant,
-) -> Result<fmt::Result, Error> {
-    let mut hrp = hrp;
-    hrp.lowercase();
-    encode_to_fmt_anycase(fmt, hrp, data, variant)
-}
-
-/// Encode a bech32 payload to an [fmt::Write], but with any case.
-/// This method is intended for implementing traits from [core::fmt] without [std].
-///
-/// See `encode_to_fmt` for meaning of errors.
-pub fn encode_to_fmt_anycase<T: AsRef<[u5]>>(
-    fmt: &mut dyn fmt::Write,
-    hrp: Hrp,
+    hrp: &Hrp,
     data: T,
     variant: Variant,
-) -> Result<fmt::Result, Error> {
+) -> fmt::Result {
     match variant {
-        Variant::Bech32 => {
-            let res = Bech32Writer::<Bech32>::new(hrp, fmt);
-            match res {
-                Ok(mut writer) => {
-                    Ok(writer.write(data.as_ref()).and_then(|_| {
-                        // Finalize manually to avoid panic on drop if write fails
-                        writer.finalize()
-                    }))
-                }
-                Err(e) => Ok(Err(e)),
-            }
-        }
-        Variant::Bech32m => {
-            let res = Bech32Writer::<Bech32m>::new(hrp, fmt);
-            match res {
-                Ok(mut writer) => {
-                    Ok(writer.write(data.as_ref()).and_then(|_| {
-                        // Finalize manually to avoid panic on drop if write fails
-                        writer.finalize()
-                    }))
-                }
-                Err(e) => Ok(Err(e)),
-            }
-        }
+        Variant::Bech32 =>
+            for c in data.as_ref().iter().copied().char_iter::<Bech32>(hrp, None) {
+                fmt.write_char(c)?;
+            },
+        Variant::Bech32m =>
+            for c in data.as_ref().iter().copied().char_iter::<Bech32>(hrp, None) {
```

Comment:
Should this be `char_iter::<Bech32m>` here?

Reply:
Yes, I"ll add a unit test also, this shouldn't have got through CI. Thanks man.